src/server/stream_family.cc

```diff
@@ -3077,12 +3077,15 @@ void XReadImpl(CmdArgList args, ReadOpts* opts, ConnectionContext* cntx) {
   for (size_t i = 0; i < xread_resp.size(); i++) {
     ShardId sid = tx->GetUniqueShardCnt() > 1 ? i : tx->GetUniqueShard();
     vector<RecordVec> sub_results = std::move(xread_resp[i]);
+
     ShardArgs shard_args = cntx->transaction->GetShardArgs(sid);
-    for (size_t j = 0; j < sub_results.size(); j++) {
+    auto shard_args_it = shard_args.begin();
```

Comment:
@dranikpg should `DCHECK_EQ(shard_args.Size(), sub_results.size());` hold? because it does not.